plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -103,11 +105,15 @@ export async function createRouter(
   );
   const taskBroker = new StorageTaskBroker(databaseTaskStore, logger);
   const actionRegistry = new TemplateActionRegistry();
-  const worker = new TaskWorker({
-    logger,
-    taskBroker,
-    actionRegistry,
-    workingDirectory,
+  const workers = new Array(taskWorkers || 1);
```

Comment:
I don't think that this would work. You'll need to fill the array first with something otherwise the `map` wont fire for `empty` elements.

`new Array(taskWorkers || 1).fill()`

Should do it, then the `.map` will run for each item.

Reply:
Run these:

```js
new Array(5).map(() => !console.log('asdas') && 'asd')
```

and 

```js
new Array(5).fill().map(() => !console.log('asdas') && 'asd')
```

And see the difference in the console :)

Reply:
But actually to be honest, maybe it's a little clearer just to do this:

const workers = [];

for (let i = 0; i < taskWorkers || 1; i++) {
  workers.push(...)
}
```

Is maybe simpler?

Reply:
Thanks! Was not aware of that nuance, updated the PR!

Reply:
@jrusso1020 I think it's better to actually just use the for loop instead sorry, do you wanna quickly just update to use that instead. It's just a little cleaner than doing `new Array(5).fill().map(() => new Worker()` :)

Reply:
@benjdlambert yeah I was in the process of doing that haha

Reply:
Updated!